File: tests/InheritedCall.cs
namespace Demo {
    public partial class Base {
        public int Foo(int a) {
            int result;
            result = a;
            return result;
        }
    }
    
    public partial class Derived : Base {
        public int Foo(int a) {
            int result;
            var request = base.Foo(a);
            result = request;
            return result;
        }
    }
}

File: tests/InheritedCall.pas
namespace Demo;

type
  Base = public class
  public
    method Foo(a: Integer): Integer; virtual;
  end;

  Derived = public class(Base)
  public
    method Foo(a: Integer): Integer; override;
  end;

implementation

method Base.Foo(a: Integer): Integer;
begin
  result := a;
end;

method Derived.Foo(a: Integer): Integer;
var request := inherited Foo(a);
begin
  result := request;
end;

end.
